app/src/main/java/es/situm/gettingstarted/drawbuilding/FloorAdapter.java

```diff
@@ -0,0 +1,97 @@
+package es.situm.gettingstarted.drawbuilding;
+
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.List;
+
+import es.situm.gettingstarted.R;
+import es.situm.sdk.model.cartography.Floor;
+
+/**
+ * Created by adriannieto on 5/10/17.
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PACKAGE_PRIVATE)
+public class FloorAdapter extends RecyclerView.Adapter<FloorAdapter.ViewHolder> {
+
+    private final OnItemClickListener onItemClickListener;
+    @Nullable
+    private Floor selected = null;
+    private List<Floor> floors;
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PACKAGE_PRIVATE)
+    FloorAdapter(List<Floor> floors, OnItemClickListener onItemClickListener) {
+        this.onItemClickListener = onItemClickListener;
+        setFloors(floors);
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PACKAGE_PRIVATE)
+    void setFloors(List<Floor> floors) {
+        this.floors = floors;
+        notifyDataSetChanged();
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
+        View view = layoutInflater.inflate(R.layout.situm_item_level, parent, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(ViewHolder holder, int position) {
+        final Floor floor = floors.get(position);
+        holder.bind(floor);
+    }
+
+    @Override
+    public int getItemCount() {
+        return floors.size();
+    }
+
+    public void select(Floor floor) {
+        selected = floor;
+        notifyDataSetChanged();
```

Comment:
Probablemente neste caso non importe, pero en conxuntos de datos maiores sería mellor ser máis específico á hora de notificar que cambiou o dataset do adapter.

Reply:
Estuve revisando los notify de RecyclerView y no encontré otro con el que sustituírlo. El que me pareció el más adecuado fue el de **notifyItemChanged()** que cambia el background del item seleccionado pero no el de los demás items, entonces acaban todos seleccionados

![Screenshot from 2022-04-12 12-08-19](https://user-images.githubusercontent.com/103040497/162936431-c5309ce7-17ba-4b9a-8d2e-26732e3e3f09.png)
.